src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -357,34 +357,30 @@ public static EnvVars getPollEnvironment(AbstractProject p, FilePath ws, Launche
             }
         }
 
-        // add env contributing actions' values from last build to environment - fixes JENKINS-22009
-        addEnvironmentContributingActionsValues(env, b);
-
         EnvVars.resolve(env);
 
         return env;
     }
 
-    private static void addEnvironmentContributingActionsValues(EnvVars env, AbstractBuild b) {
-        List<? extends Action> buildActions = b.getAllActions();
-        if (buildActions != null) {
+    public static void addEnvironmentContributingActionsValues(EnvVars env, Job project, Run b) {
+        if (b instanceof AbstractBuild) {
+            List<? extends Action> buildActions = b.getAllActions();
             for (Action action : buildActions) {
                 // most importantly, ParametersAction will be processed here (for parameterized builds)
                 if (action instanceof ParametersAction) {
                     ParametersAction envAction = (ParametersAction) action;
-                    envAction.buildEnvVars(b, env);
+                    envAction.buildEnvVars((AbstractBuild) b, env);
                 }
             }
         }
-        
         // Use the default parameter values (if any) instead of the ones from the last build
-        ParametersDefinitionProperty paramDefProp = (ParametersDefinitionProperty) b.getProject().getProperty(ParametersDefinitionProperty.class);
+        ParametersDefinitionProperty paramDefProp = (ParametersDefinitionProperty) project.getProperty(ParametersDefinitionProperty.class);
         if (paramDefProp != null) {
             for(ParameterDefinition paramDefinition : paramDefProp.getParameterDefinitions()) {
-               ParameterValue defaultValue  = paramDefinition.getDefaultParameterValue();
-               if (defaultValue != null) {
-                   defaultValue.buildEnvironment(b, env);
-               }
+                ParameterValue defaultValue  = paramDefinition.getDefaultParameterValue();
```

Comment:
Please don't modify white space, even as odd as the white space is.  Changes which modify white space tend to cause conflicts for other pull requests and increase the size of the differences.